packages/web-components/docs/feature-flags.mdx

```diff
@@ -0,0 +1,62 @@
+import { Markdown, Meta } from '@storybook/addon-docs/blocks';
+import FeatureFlag from '@docs/feature-flags.md?raw'
+
+<Meta title="Introduction/Feature Flags" />
+
+# Feature Flags
+
+[Source code](https://github.com/carbon-design-system/carbon/tree/main/packages/web-components/src/components/feature-flags)
+
+{/* prettier-ignore-start */}
+
+{/* START doctoc generated TOC please keep comment here to allow auto update */}
+
+{/* DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE */}
+
+## Table of Contents
+
+* [Current feature flags](#current-feature-flags)
+* [Turning on feature flags in web-components](#turning-on-feature-flags-in-web-components)
+* [Turning on Style feature flags in web-components](#turning-on-style-feature-flags-in-web-components)
+
+{/* END doctoc generated TOC please keep comment here to allow auto update */}
+
+{/* prettier-ignore-end */}
+
+The `@carbon/web-components` codebase ships with a number of feature flags. These feature
+flags enable new behavior and styling, allowing you to opt-in to new (and
+sometimes breaking) changes while using the current version. A feature flag may
+be configured in web-components, sass, or both.
+
+This section in the storybook documents each feature flag that is available and
+how to configure it. Folders and stories within this section in storybook show
+components with all feature flags turned on.
+
+<Markdown>{FeatureFlag}</Markdown>
+
+## Turning on feature flags in web-components
+
+Use the `<feature-flags>` web component to turn on a feature flag for a portion of your
+application. You can configure multiple feature flags at the same time by passing them as attributes or properties to the element.
+
+```jsx
+<feature-flag enableV12TileDefaultIcons="true" enableASecondFeatureFlag="true">
+  <cds-tile/>
+</feature-flag>;
+```
+
+The `<feature-flag>` component can be placed at any point in your react tree and
+will impact all children components. You can turn on feature flags for your
+entire app, or only certain pages/routes/sections of your application.
+
+## Turning on Style feature flags in web-components
+
+To enable a style-based feature flag, add the feature flag name as an attribute on the component.
+
+```jsx
+<cds-clickable-tile
+  enable-experimental-tile-contrast
+  >
+  Clickable tile
+</cds-clickable-tile>
+```
```

Comment:
Hey I have a few questions on the strategy of style flags with wc like this.

Why can't `<feature-flag>` be used to turn these on? Is there a way to read the flags from the scss directly? If someone is using `@carbon/styles` do they have to turn on in scss and here or in just one place. If there's multiple options how do they know which one to use and when?